src/core/state/networks/networks.ts

```diff
@@ -393,6 +395,28 @@ export const networkStore = createQueryStore<
       },
     ),
 
+    getChainIdsBasedOnMainnetId: createSelector(({ networks }) => {
+      return networks.backendNetworks.networks.reduce((acc, curr) => {
+        const mainnetId = +curr.mainnetId;
+        if (!acc[mainnetId]) {
+          acc[mainnetId] = [];
+        }
+        acc[mainnetId].push(+curr.id);
+        return acc;
+      }, {} as Record<number, number[]>);
+    }),
+
+    getChainsBasedOnMainnetId: createSelector(({ networks }) => {
+      return networks.backendNetworks.networks.reduce((acc, curr) => {
+        const mainnetId = +curr.mainnetId;
+        if (!acc[mainnetId]) {
+          acc[mainnetId] = [];
+        }
+        acc[mainnetId].push(curr);
+        return acc;
+      }, {} as Record<number, BackendNetwork[]>);
+    }),
+
```

Comment:
this replaces the `chainIdMap`, which we can use to construct the `chainsLabelMap`